tests/unittests/MLTest.cpp

```diff
@@ -1510,26 +1635,38 @@ TEST_P(MLTest, learnSparseLengthsWeightedSumWeights) {
   auto EH = bindings.allocate(expectedP)->getHandle();
   EH = {10, 7, 6, 3, 2};
 
-  // Allocate and store a handle to the result for testing later.
-  auto RH = bindings.allocate(result->getPlaceholder())->getHandle();
+  trainingBindings.allocate(EET_.getModule().getPlaceholders());
+  inferBindings.allocate(EEI_.getModule().getPlaceholders());
+  bindings.copyTrainedWeights(trainingBindings);
+  bindings.copyToTarget("dataP", trainingBindings);
+  bindings.copyToTarget("indicesP", trainingBindings);
+  bindings.copyToTarget("weightsP", trainingBindings);
+  bindings.copyToTarget("lengthsP", trainingBindings);
+  bindings.copyToTarget("expectedP", trainingBindings);
+  auto *res = inferBindings.get(EEI_.getModule().getPlaceholderByName("save"));
 
   // Train the network.
-  Function *trainingGradientFunction = glow::differentiate(F, TC);
-  EE_.compile(CompilationMode::Train, trainingGradientFunction);
+  glow::differentiate(F, TC);
+  EET_.compile(CompilationMode::Train);
 
   const size_t numIterations = 1000;
 
   for (size_t i = 0; i < numIterations; ++i) {
-    EE_.run(bindings);
+    EET_.run(trainingBindings, "SparseLengthsWeightedSum_grad");
   }
-
+  trainingBindings.copyTrainedWeights(inferBindings);
+  trainingBindings.copyToTarget("dataP", inferBindings);
+  trainingBindings.copyToTarget("indicesP", inferBindings);
+  trainingBindings.copyToTarget("weightsP", inferBindings);
+  trainingBindings.copyToTarget("lengthsP", inferBindings);
+  trainingBindings.copyToTarget("expectedP", inferBindings);
   // Switch to inference mode and run the network.
-  EE_.compile(CompilationMode::Infer, F);
-  EE_.run(bindings);
+  EEI_.compile(CompilationMode::Infer);
+  EEI_.run(inferBindings, "SparseLengthsWeightedSum");
 
   // Make sure that the network output matches expectations after training.
   for (size_t j = 0; j < EH.size(); ++j) {
-    EXPECT_NEAR(RH.raw(j), EH.raw(j), 0.02);
+    EXPECT_NEAR(res->getHandle().raw(j), EH.raw(j), 0.02);
```

Comment:
nit: We generally avoid getting handles inside loops. Instead get it outside the loop and reuse.